sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/AdaptiveSparkPlanExec.scala

```diff
@@ -79,6 +81,13 @@ case class AdaptiveSparkPlanExec(
     case _ => logDebug(_)
   }
 
+  override lazy val metrics = Map(
+    "num broadcast join conversions" ->
```

Comment:
these are not really metrics. how about `logDebug`?

Reply:
Making these proper metrics makes them show up in REST endpoints:  http://localhost:4040/api/v1/applications/local-1628523514017/sql/27/ (and equivalent History Server Endpoint) which enables automation to be build.
That wouldn't easily work if this is just logged.
Why are these less "metrics" than number of partitions, for example?

Reply:
@cloud-fan do you have any more thoughts on this?

Reply:
I think these are useful "numbers", but it's a bit weird to see them in the UI (compared to low-level metrics like bytesRead, numPartitions). I think we need a better place to put these numbers.

cc @maryannxue @hvanhovell @viirya

Reply:
Hello,
@cloud-fan, @maryannxue @hvanhovell @viirya
do any of you have  any suggesting on how to proceed?
It would also be useful to add "reOptimize duration" to track total `reOptimize()` time and "generate explainString duration" to track `context.qe.explainString` which can be noticeable for large plans.
These are also not exactly the same as other "metrics" but `AdaptiveSparkPlanExec` is also quite different from other nodes.